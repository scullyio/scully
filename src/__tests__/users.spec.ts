import {readFileSync} from 'fs';
import {join} from 'path';
import {replaceIndexNG} from '../test-config.helper';

describe('JsonPlugin: test user List', () => {
  it('Check clean blog index by scully', () => {
    const index: string = readFileSync(
      join(__dirname, '../../dist/static/user/index.html'),
      'UTF8'
    ).toString();
    const cleanIndex = replaceIndexNG(index);
    expect(cleanIndex).toMatchSnapshot();
  });
});

describe('TransferState: test user', () => {
  it('Check clean blog index by scully', () => {
    const index: string = readFileSync(
      join(__dirname, '../../dist/static/user/1/index.html'),
      'UTF8'
    ).toString();
    const cleanIndex = replaceIndexNG(index);
    expect(cleanIndex).toMatchSnapshot();
  });
});

describe('TransferState: test user post', () => {
  it('Check clean blog index by scully', () => {
    const index: string = readFileSync(
      join(__dirname, '../../dist/static/user/1/post/1/index.html'),
      'UTF8'
    ).toString();
    const cleanIndex = replaceIndexNG(index);
    expect(cleanIndex).toMatchSnapshot();
  });
});
