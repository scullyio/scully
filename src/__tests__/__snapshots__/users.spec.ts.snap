// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JsonPlugin: test user List Check clean blog index by scully 1`] = `
"<!DOCTYPE html><html lang=\\"en\\"><head>
    <meta charset=\\"utf-8\\">
    <title>SampleBlog</title>
    <base href=\\"/\\">
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
    <link rel=\\"icon\\" type=\\"image/x-icon\\" href=\\"favicon.ico\\">
  <link rel=\\"stylesheet\\" href=\\"styles.css\\"><script id=\\"scully-transfer-state\\">window['scully-transfer-state']=/** ___SCULLY_STATE_START___ */{\\"users\\":[{\\"id\\":1,\\"name\\":\\"Leanne Graham\\",\\"username\\":\\"Bret\\",\\"email\\":\\"Sincere@april.biz\\",\\"address\\":{\\"street\\":\\"Kulas Light\\",\\"suite\\":\\"Apt. 556\\",\\"city\\":\\"Gwenborough\\",\\"zipcode\\":\\"92998-3874\\",\\"geo\\":{\\"lat\\":\\"-37.3159\\",\\"lng\\":\\"81.1496\\"}},\\"phone\\":\\"1-770-736-8031 x56442\\",\\"website\\":\\"hildegard.org\\",\\"company\\":{\\"name\\":\\"Romaguera-Crona\\",\\"catchPhrase\\":\\"Multi-layered client-server neural-net\\",\\"bs\\":\\"harness real-time e-markets\\"}},{\\"id\\":2,\\"name\\":\\"Ervin Howell\\",\\"username\\":\\"Antonette\\",\\"email\\":\\"Shanna@melissa.tv\\",\\"address\\":{\\"street\\":\\"Victor Plains\\",\\"suite\\":\\"Suite 879\\",\\"city\\":\\"Wisokyburgh\\",\\"zipcode\\":\\"90566-7771\\",\\"geo\\":{\\"lat\\":\\"-43.9509\\",\\"lng\\":\\"-34.4618\\"}},\\"phone\\":\\"010-692-6593 x09125\\",\\"website\\":\\"anastasia.net\\",\\"company\\":{\\"name\\":\\"Deckow-Crist\\",\\"catchPhrase\\":\\"Proactive didactic contingency\\",\\"bs\\":\\"synergize scalable supply-chains\\"}},{\\"id\\":3,\\"name\\":\\"Clementine Bauch\\",\\"username\\":\\"Samantha\\",\\"email\\":\\"Nathan@yesenia.net\\",\\"address\\":{\\"street\\":\\"Douglas Extension\\",\\"suite\\":\\"Suite 847\\",\\"city\\":\\"McKenziehaven\\",\\"zipcode\\":\\"59590-4157\\",\\"geo\\":{\\"lat\\":\\"-68.6102\\",\\"lng\\":\\"-47.0653\\"}},\\"phone\\":\\"1-463-123-4447\\",\\"website\\":\\"ramiro.info\\",\\"company\\":{\\"name\\":\\"Romaguera-Jacobson\\",\\"catchPhrase\\":\\"Face to face bifurcated interface\\",\\"bs\\":\\"e-enable strategic applications\\"}},{\\"id\\":4,\\"name\\":\\"Patricia Lebsack\\",\\"username\\":\\"Karianne\\",\\"email\\":\\"Julianne.OConner@kory.org\\",\\"address\\":{\\"street\\":\\"Hoeger Mall\\",\\"suite\\":\\"Apt. 692\\",\\"city\\":\\"South Elvis\\",\\"zipcode\\":\\"53919-4257\\",\\"geo\\":{\\"lat\\":\\"29.4572\\",\\"lng\\":\\"-164.2990\\"}},\\"phone\\":\\"493-170-9623 x156\\",\\"website\\":\\"kale.biz\\",\\"company\\":{\\"name\\":\\"Robel-Corkery\\",\\"catchPhrase\\":\\"Multi-tiered zero tolerance productivity\\",\\"bs\\":\\"transition cutting-edge web services\\"}},{\\"id\\":5,\\"name\\":\\"Chelsey Dietrich\\",\\"username\\":\\"Kamren\\",\\"email\\":\\"Lucio_Hettinger@annie.ca\\",\\"address\\":{\\"street\\":\\"Skiles Walks\\",\\"suite\\":\\"Suite 351\\",\\"city\\":\\"Roscoeview\\",\\"zipcode\\":\\"33263\\",\\"geo\\":{\\"lat\\":\\"-31.8129\\",\\"lng\\":\\"62.5342\\"}},\\"phone\\":\\"(254)954-1289\\",\\"website\\":\\"demarco.info\\",\\"company\\":{\\"name\\":\\"Keebler LLC\\",\\"catchPhrase\\":\\"User-centric fault-tolerant solution\\",\\"bs\\":\\"revolutionize end-to-end systems\\"}},{\\"id\\":6,\\"name\\":\\"Mrs. Dennis Schulist\\",\\"username\\":\\"Leopoldo_Corkery\\",\\"email\\":\\"Karley_Dach@jasper.info\\",\\"address\\":{\\"street\\":\\"Norberto Crossing\\",\\"suite\\":\\"Apt. 950\\",\\"city\\":\\"South Christy\\",\\"zipcode\\":\\"23505-1337\\",\\"geo\\":{\\"lat\\":\\"-71.4197\\",\\"lng\\":\\"71.7478\\"}},\\"phone\\":\\"1-477-935-8478 x6430\\",\\"website\\":\\"ola.org\\",\\"company\\":{\\"name\\":\\"Considine-Lockman\\",\\"catchPhrase\\":\\"Synchronised bottom-line interface\\",\\"bs\\":\\"e-enable innovative applications\\"}},{\\"id\\":7,\\"name\\":\\"Kurtis Weissnat\\",\\"username\\":\\"Elwyn.Skiles\\",\\"email\\":\\"Telly.Hoeger@billy.biz\\",\\"address\\":{\\"street\\":\\"Rex Trail\\",\\"suite\\":\\"Suite 280\\",\\"city\\":\\"Howemouth\\",\\"zipcode\\":\\"58804-1099\\",\\"geo\\":{\\"lat\\":\\"24.8918\\",\\"lng\\":\\"21.8984\\"}},\\"phone\\":\\"210.067.6132\\",\\"website\\":\\"elvis.io\\",\\"company\\":{\\"name\\":\\"Johns Group\\",\\"catchPhrase\\":\\"Configurable multimedia task-force\\",\\"bs\\":\\"generate enterprise e-tailers\\"}},{\\"id\\":8,\\"name\\":\\"Nicholas Runolfsdottir V\\",\\"username\\":\\"Maxime_Nienow\\",\\"email\\":\\"Sherwood@rosamond.me\\",\\"address\\":{\\"street\\":\\"Ellsworth Summit\\",\\"suite\\":\\"Suite 729\\",\\"city\\":\\"Aliyaview\\",\\"zipcode\\":\\"45169\\",\\"geo\\":{\\"lat\\":\\"-14.3990\\",\\"lng\\":\\"-120.7677\\"}},\\"phone\\":\\"586.493.6943 x140\\",\\"website\\":\\"jacynthe.com\\",\\"company\\":{\\"name\\":\\"Abernathy Group\\",\\"catchPhrase\\":\\"Implemented secondary concept\\",\\"bs\\":\\"e-enable extensible e-tailers\\"}},{\\"id\\":9,\\"name\\":\\"Glenna Reichert\\",\\"username\\":\\"Delphine\\",\\"email\\":\\"Chaim_McDermott@dana.io\\",\\"address\\":{\\"street\\":\\"Dayna Park\\",\\"suite\\":\\"Suite 449\\",\\"city\\":\\"Bartholomebury\\",\\"zipcode\\":\\"76495-3109\\",\\"geo\\":{\\"lat\\":\\"24.6463\\",\\"lng\\":\\"-168.8889\\"}},\\"phone\\":\\"(775)976-6794 x41206\\",\\"website\\":\\"conrad.com\\",\\"company\\":{\\"name\\":\\"Yost and Sons\\",\\"catchPhrase\\":\\"Switchable contextually-based project\\",\\"bs\\":\\"aggregate real-time technologies\\"}},{\\"id\\":10,\\"name\\":\\"Clementina DuBuque\\",\\"username\\":\\"Moriah.Stanton\\",\\"email\\":\\"Rey.Padberg@karina.biz\\",\\"address\\":{\\"street\\":\\"Kattie Turnpike\\",\\"suite\\":\\"Suite 198\\",\\"city\\":\\"Lebsackbury\\",\\"zipcode\\":\\"31428-2261\\",\\"geo\\":{\\"lat\\":\\"-38.2386\\",\\"lng\\":\\"57.2232\\"}},\\"phone\\":\\"024-648-3804\\",\\"website\\":\\"ambrose.net\\",\\"company\\":{\\"name\\":\\"Hoeger LLC\\",\\"catchPhrase\\":\\"Centralized empowering task-force\\",\\"bs\\":\\"target end-to-end models\\"}}]}/** ___SCULLY_STATE_END___ */</script><style>[]{display:grid;height:100vh;grid-template-rows:60px 1fr 60px}[] > header[]{display:grid;background-color:#4169e1;color:#f5f5f5;margin:0;padding:0 10px;grid-template-columns:1fr 40px;place-items:center center}footer[]   h3[], header[]   h1[]{margin:0}[] > footer[]{display:grid;background-color:#4169e1;color:#f5f5f5;margin:0;padding-right:10px;justify-content:right;align-content:center}main[]{background-color:#f5f5f5;color:#040c24;padding:10px}</style><script charset=\\"utf-8\\" src=\\"user-user-module-es2015.js\\"></script><script charset=\\"utf-8\\" src=\\"pagenotfound-pagenotfound-module-es2015.js\\"></script><style></style><script>window['ScullyIO']='generated';</script></head>

  <body scully-version=\\"0.0.0\\">
    <app-root =\\"\\" > Scully demo blog app! <small =\\"\\">rendering inside scully</small></h1><a =\\"\\" href=\\"/home\\">üè†</a></header><main =\\"\\"><router-outlet =\\"\\"></router-outlet><app-users =\\"\\"><h1 =\\"\\">Users</h1><section =\\"\\"><table =\\"\\"><thead =\\"\\"><th =\\"\\">Id</th><th =\\"\\">Username</th></thead><tbody =\\"\\"><tr =\\"\\"><td =\\"\\">1</td><td =\\"\\"><a =\\"\\" href=\\"/user/1\\"> Leanne Graham</a></td></tr><tr =\\"\\"><td =\\"\\">2</td><td =\\"\\"><a =\\"\\" href=\\"/user/2\\"> Ervin Howell</a></td></tr><tr =\\"\\"><td =\\"\\">3</td><td =\\"\\"><a =\\"\\" href=\\"/user/3\\"> Clementine Bauch</a></td></tr><tr =\\"\\"><td =\\"\\">4</td><td =\\"\\"><a =\\"\\" href=\\"/user/4\\"> Patricia Lebsack</a></td></tr><tr =\\"\\"><td =\\"\\">5</td><td =\\"\\"><a =\\"\\" href=\\"/user/5\\"> Chelsey Dietrich</a></td></tr><tr =\\"\\"><td =\\"\\">6</td><td =\\"\\"><a =\\"\\" href=\\"/user/6\\"> Mrs. Dennis Schulist</a></td></tr><tr =\\"\\"><td =\\"\\">7</td><td =\\"\\"><a =\\"\\" href=\\"/user/7\\"> Kurtis Weissnat</a></td></tr><tr =\\"\\"><td =\\"\\">8</td><td =\\"\\"><a =\\"\\" href=\\"/user/8\\"> Nicholas Runolfsdottir V</a></td></tr><tr =\\"\\"><td =\\"\\">9</td><td =\\"\\"><a =\\"\\" href=\\"/user/9\\"> Glenna Reichert</a></td></tr><tr =\\"\\"><td =\\"\\">10</td><td =\\"\\"><a =\\"\\" href=\\"/user/10\\"> Clementina DuBuque</a></td></tr><!----></tbody></table></section><!----></app-users><!----></main><footer =\\"\\"><h3 =\\"\\">Made with ‚ù§Ô∏è <strong =\\"\\">@HeroDevs</strong></h3></footer></app-root>
  <script src=\\"runtime-es2015.js\\" type=\\"module\\"></script><script src=\\"runtime-es5.js\\" nomodule=\\"\\" defer=\\"\\"></script><script src=\\"polyfills-es5.js\\" nomodule=\\"\\" defer=\\"\\"></script><script src=\\"polyfills-es2015.js\\" type=\\"module\\"></script><script src=\\"main-es2015.js\\" type=\\"module\\"></script><script src=\\"main-es5.js\\" nomodule=\\"\\" defer=\\"\\"></script>

</body></html>"
`;

exports[`TransferState: test user Check clean blog index by scully 1`] = `
"<!DOCTYPE html><html lang=\\"en\\"><head>
    <meta charset=\\"utf-8\\">
    <title>SampleBlog</title>
    <base href=\\"/\\">
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
    <link rel=\\"icon\\" type=\\"image/x-icon\\" href=\\"favicon.ico\\">
  <link rel=\\"stylesheet\\" href=\\"styles.css\\"><script id=\\"scully-transfer-state\\">window['scully-transfer-state']=/** ___SCULLY_STATE_START___ */{\\"posts\\":[{\\"userId\\":1,\\"id\\":1,\\"title\\":\\"sunt aut facere repellat provident occaecati excepturi optio reprehenderit\\",\\"body\\":\\"quia et suscipit\\\\nsuscipit recusandae consequuntur expedita et cum\\\\nreprehenderit molestiae ut ut quas totam\\\\nnostrum rerum est autem sunt rem eveniet architecto\\"},{\\"userId\\":1,\\"id\\":2,\\"title\\":\\"qui est esse\\",\\"body\\":\\"est rerum tempore vitae\\\\nsequi sint nihil reprehenderit dolor beatae ea dolores neque\\\\nfugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis\\\\nqui aperiam non debitis possimus qui neque nisi nulla\\"},{\\"userId\\":1,\\"id\\":3,\\"title\\":\\"ea molestias quasi exercitationem repellat qui ipsa sit aut\\",\\"body\\":\\"et iusto sed quo iure\\\\nvoluptatem occaecati omnis eligendi aut ad\\\\nvoluptatem doloribus vel accusantium quis pariatur\\\\nmolestiae porro eius odio et labore et velit aut\\"},{\\"userId\\":1,\\"id\\":4,\\"title\\":\\"eum et est occaecati\\",\\"body\\":\\"ullam et saepe reiciendis voluptatem adipisci\\\\nsit amet autem assumenda provident rerum culpa\\\\nquis hic commodi nesciunt rem tenetur doloremque ipsam iure\\\\nquis sunt voluptatem rerum illo velit\\"},{\\"userId\\":1,\\"id\\":5,\\"title\\":\\"nesciunt quas odio\\",\\"body\\":\\"repudiandae veniam quaerat sunt sed\\\\nalias aut fugiat sit autem sed est\\\\nvoluptatem omnis possimus esse voluptatibus quis\\\\nest aut tenetur dolor neque\\"},{\\"userId\\":1,\\"id\\":6,\\"title\\":\\"dolorem eum magni eos aperiam quia\\",\\"body\\":\\"ut aspernatur corporis harum nihil quis provident sequi\\\\nmollitia nobis aliquid molestiae\\\\nperspiciatis et ea nemo ab reprehenderit accusantium quas\\\\nvoluptate dolores velit et doloremque molestiae\\"},{\\"userId\\":1,\\"id\\":7,\\"title\\":\\"magnam facilis autem\\",\\"body\\":\\"dolore placeat quibusdam ea quo vitae\\\\nmagni quis enim qui quis quo nemo aut saepe\\\\nquidem repellat excepturi ut quia\\\\nsunt ut sequi eos ea sed quas\\"},{\\"userId\\":1,\\"id\\":8,\\"title\\":\\"dolorem dolore est ipsam\\",\\"body\\":\\"dignissimos aperiam dolorem qui eum\\\\nfacilis quibusdam animi sint suscipit qui sint possimus cum\\\\nquaerat magni maiores excepturi\\\\nipsam ut commodi dolor voluptatum modi aut vitae\\"},{\\"userId\\":1,\\"id\\":9,\\"title\\":\\"nesciunt iure omnis dolorem tempora et accusantium\\",\\"body\\":\\"consectetur animi nesciunt iure dolore\\\\nenim quia ad\\\\nveniam autem ut quam aut nobis\\\\net est aut quod aut provident voluptas autem voluptas\\"},{\\"userId\\":1,\\"id\\":10,\\"title\\":\\"optio molestias id quia eum\\",\\"body\\":\\"quo et expedita modi cum officia vel magni\\\\ndoloribus qui repudiandae\\\\nvero nisi sit\\\\nquos veniam quod sed accusamus veritatis error\\"}],\\"user\\":{\\"id\\":1,\\"name\\":\\"Leanne Graham\\",\\"username\\":\\"Bret\\",\\"email\\":\\"Sincere@april.biz\\",\\"address\\":{\\"street\\":\\"Kulas Light\\",\\"suite\\":\\"Apt. 556\\",\\"city\\":\\"Gwenborough\\",\\"zipcode\\":\\"92998-3874\\",\\"geo\\":{\\"lat\\":\\"-37.3159\\",\\"lng\\":\\"81.1496\\"}},\\"phone\\":\\"1-770-736-8031 x56442\\",\\"website\\":\\"hildegard.org\\",\\"company\\":{\\"name\\":\\"Romaguera-Crona\\",\\"catchPhrase\\":\\"Multi-layered client-server neural-net\\",\\"bs\\":\\"harness real-time e-markets\\"}}}/** ___SCULLY_STATE_END___ */</script><style>[]{display:grid;height:100vh;grid-template-rows:60px 1fr 60px}[] > header[]{display:grid;background-color:#4169e1;color:#f5f5f5;margin:0;padding:0 10px;grid-template-columns:1fr 40px;place-items:center center}footer[]   h3[], header[]   h1[]{margin:0}[] > footer[]{display:grid;background-color:#4169e1;color:#f5f5f5;margin:0;padding-right:10px;justify-content:right;align-content:center}main[]{background-color:#f5f5f5;color:#040c24;padding:10px}</style><script charset=\\"utf-8\\" src=\\"user-user-module-es2015.js\\"></script><script charset=\\"utf-8\\" src=\\"pagenotfound-pagenotfound-module-es2015.js\\"></script><style>a[]{display:inline-block;padding:5px;margin:10px 3px;width:100px;border-radius:4px;background-color:#4169e1;color:#fff;text-align:center;text-decoration:none;box-shadow:0 1px 3px rgba(0,0,0,.12),0 1px 2px rgba(0,0,0,.24);transition:all .3s cubic-bezier(.25,.8,.25,1)}a[]:hover{box-shadow:0 14px 28px rgba(0,0,0,.25),0 10px 10px rgba(0,0,0,.22)}a[]:active{box-shadow:0 1px 3px rgba(0,0,0,.12),0 1px 2px rgba(0,0,0,.24)}</style><style>#user[] {
    display: grid;
    grid-template-columns: 10rem 1fr;
    grid-template-rows: repeat(4, 1rem);
  }

  #user[]   .label[] {
    text-align: right;
  }

  #user[]   .label[]::after {
    content: ':';
  }

  strong[] {
    display: inline-block;
    padding: 5px;
    border-radius: 5px;
    background-color: #afafaf;
    margin: 10px;
  }
  #sub[] {
    padding: 20px;
  }</style><style></style><script>window['ScullyIO']='generated';</script></head>

  <body scully-version=\\"0.0.0\\">
    <app-root =\\"\\" > Scully demo blog app! <small =\\"\\">rendering inside scully</small></h1><a =\\"\\" href=\\"/home\\">üè†</a></header><main =\\"\\"><router-outlet =\\"\\"></router-outlet><app-user =\\"\\"><section =\\"\\"><h4 =\\"\\">User Data</h4><section =\\"\\" id=\\"user\\"><label =\\"\\">Id</label><p =\\"\\">1</p><label =\\"\\">Name</label><p =\\"\\">Leanne Graham</p><label =\\"\\">email</label><p =\\"\\">Sincere@april.biz</p><label =\\"\\">company</label><p =\\"\\">Romaguera-Crona</p></section><strong =\\"\\">Multi-layered client-server neural-net</strong></section><!----><!----><a =\\"\\" href=\\"/user/1\\">Previous (1)</a><a =\\"\\" href=\\"/user/2\\">Next (2)</a><hr =\\"\\"><section =\\"\\" id=\\"sub\\"><router-outlet =\\"\\"></router-outlet><app-posts =\\"\\"><h4 =\\"\\">User posts</h4><section =\\"\\"><ul =\\"\\"><li =\\"\\"><a =\\"\\" href=\\"/user/1/post/1\\">sunt aut facere repellat provident occaecati excepturi optio reprehenderit</a></li><li =\\"\\"><a =\\"\\" href=\\"/user/1/post/2\\">qui est esse</a></li><li =\\"\\"><a =\\"\\" href=\\"/user/1/post/3\\">ea molestias quasi exercitationem repellat qui ipsa sit aut</a></li><li =\\"\\"><a =\\"\\" href=\\"/user/1/post/4\\">eum et est occaecati</a></li><li =\\"\\"><a =\\"\\" href=\\"/user/1/post/5\\">nesciunt quas odio</a></li><li =\\"\\"><a =\\"\\" href=\\"/user/1/post/6\\">dolorem eum magni eos aperiam quia</a></li><li =\\"\\"><a =\\"\\" href=\\"/user/1/post/7\\">magnam facilis autem</a></li><li =\\"\\"><a =\\"\\" href=\\"/user/1/post/8\\">dolorem dolore est ipsam</a></li><li =\\"\\"><a =\\"\\" href=\\"/user/1/post/9\\">nesciunt iure omnis dolorem tempora et accusantium</a></li><li =\\"\\"><a =\\"\\" href=\\"/user/1/post/10\\">optio molestias id quia eum</a></li><!----></ul></section><!----></app-posts><!----></section></app-user><!----></main><footer =\\"\\"><h3 =\\"\\">Made with ‚ù§Ô∏è <strong =\\"\\">@HeroDevs</strong></h3></footer></app-root>
  <script src=\\"runtime-es2015.js\\" type=\\"module\\"></script><script src=\\"runtime-es5.js\\" nomodule=\\"\\" defer=\\"\\"></script><script src=\\"polyfills-es5.js\\" nomodule=\\"\\" defer=\\"\\"></script><script src=\\"polyfills-es2015.js\\" type=\\"module\\"></script><script src=\\"main-es2015.js\\" type=\\"module\\"></script><script src=\\"main-es5.js\\" nomodule=\\"\\" defer=\\"\\"></script>

</body></html>"
`;

exports[`TransferState: test user post Check clean blog index by scully 1`] = `
"<!DOCTYPE html><html lang=\\"en\\"><head>
    <meta charset=\\"utf-8\\">
    <title>SampleBlog</title>
    <base href=\\"/\\">
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
    <link rel=\\"icon\\" type=\\"image/x-icon\\" href=\\"favicon.ico\\">
  <link rel=\\"stylesheet\\" href=\\"styles.css\\"><script id=\\"scully-transfer-state\\">window['scully-transfer-state']=/** ___SCULLY_STATE_START___ */{\\"user\\":{\\"id\\":1,\\"name\\":\\"Leanne Graham\\",\\"username\\":\\"Bret\\",\\"email\\":\\"Sincere@april.biz\\",\\"address\\":{\\"street\\":\\"Kulas Light\\",\\"suite\\":\\"Apt. 556\\",\\"city\\":\\"Gwenborough\\",\\"zipcode\\":\\"92998-3874\\",\\"geo\\":{\\"lat\\":\\"-37.3159\\",\\"lng\\":\\"81.1496\\"}},\\"phone\\":\\"1-770-736-8031 x56442\\",\\"website\\":\\"hildegard.org\\",\\"company\\":{\\"name\\":\\"Romaguera-Crona\\",\\"catchPhrase\\":\\"Multi-layered client-server neural-net\\",\\"bs\\":\\"harness real-time e-markets\\"}},\\"post\\":{\\"userId\\":1,\\"id\\":1,\\"title\\":\\"sunt aut facere repellat provident occaecati excepturi optio reprehenderit\\",\\"body\\":\\"quia et suscipit\\\\nsuscipit recusandae consequuntur expedita et cum\\\\nreprehenderit molestiae ut ut quas totam\\\\nnostrum rerum est autem sunt rem eveniet architecto\\"}}/** ___SCULLY_STATE_END___ */</script><style>[]{display:grid;height:100vh;grid-template-rows:60px 1fr 60px}[] > header[]{display:grid;background-color:#4169e1;color:#f5f5f5;margin:0;padding:0 10px;grid-template-columns:1fr 40px;place-items:center center}footer[]   h3[], header[]   h1[]{margin:0}[] > footer[]{display:grid;background-color:#4169e1;color:#f5f5f5;margin:0;padding-right:10px;justify-content:right;align-content:center}main[]{background-color:#f5f5f5;color:#040c24;padding:10px}</style><script charset=\\"utf-8\\" src=\\"user-user-module-es2015.js\\"></script><script charset=\\"utf-8\\" src=\\"pagenotfound-pagenotfound-module-es2015.js\\"></script><style>a[]{display:inline-block;padding:5px;margin:10px 3px;width:100px;border-radius:4px;background-color:#4169e1;color:#fff;text-align:center;text-decoration:none;box-shadow:0 1px 3px rgba(0,0,0,.12),0 1px 2px rgba(0,0,0,.24);transition:all .3s cubic-bezier(.25,.8,.25,1)}a[]:hover{box-shadow:0 14px 28px rgba(0,0,0,.25),0 10px 10px rgba(0,0,0,.22)}a[]:active{box-shadow:0 1px 3px rgba(0,0,0,.12),0 1px 2px rgba(0,0,0,.24)}</style><style>#user[] {
    display: grid;
    grid-template-columns: 10rem 1fr;
    grid-template-rows: repeat(4, 1rem);
  }

  #user[]   .label[] {
    text-align: right;
  }

  #user[]   .label[]::after {
    content: ':';
  }

  strong[] {
    display: inline-block;
    padding: 5px;
    border-radius: 5px;
    background-color: #afafaf;
    margin: 10px;
  }
  #sub[] {
    padding: 20px;
  }</style><style></style><script>window['ScullyIO']='generated';</script></head>

  <body scully-version=\\"0.0.0\\">
    <app-root =\\"\\" > Scully demo blog app! <small =\\"\\">rendering inside scully</small></h1><a =\\"\\" href=\\"/home\\">üè†</a></header><main =\\"\\"><router-outlet =\\"\\"></router-outlet><app-user =\\"\\"><section =\\"\\"><h4 =\\"\\">User Data</h4><section =\\"\\" id=\\"user\\"><label =\\"\\">Id</label><p =\\"\\">1</p><label =\\"\\">Name</label><p =\\"\\">Leanne Graham</p><label =\\"\\">email</label><p =\\"\\">Sincere@april.biz</p><label =\\"\\">company</label><p =\\"\\">Romaguera-Crona</p></section><strong =\\"\\">Multi-layered client-server neural-net</strong></section><!----><!----><a =\\"\\" href=\\"/user/1\\">Previous (1)</a><a =\\"\\" href=\\"/user/2\\">Next (2)</a><hr =\\"\\"><section =\\"\\" id=\\"sub\\"><router-outlet =\\"\\"></router-outlet><app-post =\\"\\"><h4 =\\"\\">Post details</h4><section =\\"\\"><h4 =\\"\\">sunt aut facere repellat provident occaecati excepturi optio reprehenderit</h4><p =\\"\\">quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto</p></section><!----></app-post><!----></section></app-user><!----></main><footer =\\"\\"><h3 =\\"\\">Made with ‚ù§Ô∏è <strong =\\"\\">@HeroDevs</strong></h3></footer></app-root>
  <script src=\\"runtime-es2015.js\\" type=\\"module\\"></script><script src=\\"runtime-es5.js\\" nomodule=\\"\\" defer=\\"\\"></script><script src=\\"polyfills-es5.js\\" nomodule=\\"\\" defer=\\"\\"></script><script src=\\"polyfills-es2015.js\\" type=\\"module\\"></script><script src=\\"main-es2015.js\\" type=\\"module\\"></script><script src=\\"main-es5.js\\" nomodule=\\"\\" defer=\\"\\"></script>

</body></html>"
`;
